### PR TITLE
Fixed the type issue

### LINKED ISSUE #227: InDims type error
This code fails:
```julia
using EarthDataLab, YAXArrays 

### Cube handle 

cube = esdd() # Telekom object storage accessible from everywhere

### subsetting 

csub = cube[
  var = [:Rg, :air_temperature_2m, :bare_soil_evaporation, :evaporation, :evaporative_stress, :fapar_tip, :fractional_snow_cover, :gross_primary_productivity, :interception_loss,  :land_surface_temperature, :latent_energy, :leaf_area_index, :max_air_temperature_2m, :min_air_temperature_2m, :net_radiation, :open_water_evaporation, :ozone, :par, :pardiff, :potential_evaporation, :precipitation, :precipitation_era5, :psurf, :radiation_era5, :root_moisture, :sensible_heat, :snow_sublimation, :snow_water_equivalent, :surface_moisture, :transpiration, :water_mask, :water_vapour],
  time = 2000:2001
]
### aggregate in time

using StatsBase

cmean = mapslices(mean ∘ skipmissing, csub; dims="Time")
```
The error is:
```
ERROR: MethodError: no method matching InDims(::Tuple{YAXArrays.Cubes.Axes.ByName}, ::Type{Array}, ::Tuple{YAXArrays.DAT.AllMissing})
Closest candidates are:
  InDims(::Tuple, ::Any, ::Tuple, ::Any) at ~/.julia/packages/YAXArrays/Fe7F8/src/DAT/registration.jl:77
  InDims(::Any, ::Any, ::Any, ::Any) at ~/.julia/packages/YAXArrays/Fe7F8/src/DAT/registration.jl:77
Stacktrace:
 [1] (::YAXArrays.DAT.var"#39#44"{InDims, Vector{YAXArrays.Cubes.Axes.ByName}})(c::YAXArray{Union{Missing, Float32}, 3, DiskArrays.SubDiskArray{Union{Missing, Float32}, 3}, Vector{CubeAxis}})
   @ YAXArrays.DAT ~/.julia/packages/YAXArrays/Fe7F8/src/DAT/DAT.jl:338
    [...]
 [7] top-level scope
   @ ~/.../yaxtest.jl:17
```

### PR DESCRIPTION
**Pull request by Qfl3x**
Attempt to fix  #227. Error no longer occurs.

### CODE REVIEW

**Comment by Qfl3x** [comment]
If there ellipsis is unnecessary, I suggest removing it entirely as I did. (Only use the `catch`  line)

File: src/DAT/registration.jl [hunk 1]
```diff
@@ -80,15 +80,23 @@ mutable struct InDims
...
-    axisdesc::Union{String,CubeAxis,Symbol,MovingWindow}...;
+    axisdesc::Union{String,CubeAxis,Symbol,MovingWindow,AxisDescriptor}...;
+    artype = Array,
+    filter = AllMissing(),
+    window_oob_value = missing,
+)
+    descs = get_descriptor.(axisdesc)
+    InDims(descs, artype, getprocfilter(filter), window_oob_value)
+end
+function InDimsUser(
```
Comment (felixcremer) [inline src/DAT/registration.jl:91]: What is InDimsUser? 
Shouldn't it be enough to change line 83? The two functions seem to have the same code and with line 83 we would get there as well via multiple dispatch.

**Comment by felixcremer** [comment]
Thank you for your PR. I am not sure, whether the try catch construct is necessary there, because for me it seems that we are missing a method of InDims with the Tuple and an window_oob_value. I am going to have another look tomorrow.

File: src/DAT/registration.jl (hunk 1 above)
Comment (Qfl3x) [reply src/DAT/registration.jl:91]: Forgot to clean it, That's just my debug function to make sure it's not something deeper. Please ignore

**Comment by felixcremer** [comment]
Could you add a test case for the mapslices of a dataset? This should go into the test/Datasets/datasets.jl file. You could for example test the mean via mapslices of a Dataset dataset vs the mapslices(mean, yax.data, dimsnumber) for all yax in dataset.cubes

**Comment by Qfl3x** [comment]
I removed the try, catch block and implemented a much cleaner solution using Multiple Dispatch.

**Comment by Qfl3x** [comment]
It can be cleaned further and implement both the `InDims` into one function. However, it depends what you want to do with tuples of different types (ex: a tuple of 2 `AxisDescriptors`). Currently, this probably crashes.

**Comment by lazarusA** [comment]
@felixcremer looks clean. And nothing else seems to be broken.

**Review by meggart — APPROVED** [review]

**Pull request merged**